shipyard_blueprints/coalesce/shipyard_coalesce/coalesce.py

```diff
@@ -105,54 +116,63 @@ def determine_sync_status(self, run_counter: int) -> int:
         Returns:
             int: An exit code for the Shipyard Application
         """
-        response = self.get_run_status(run_counter)
-        json = response.json()
-        if response.status_code == 200:
-            status = json.get("runStatus")
-            # go through the statuses and return the appropriate exit code
-            # documentation does not provide options for status aside from completed
-            if status == "completed":
-                self.logger.info("Status: completed")
-                return self.EXIT_CODE_FINAL_STATUS_COMPLETED
-            elif status == "pending":
-                self.logger.info("Status: pending")
-                return self.EXIT_CODE_FINAL_STATUS_PENDING
-            elif status == "running":
-                self.logger.info("Status: running")
-                return self.EXIT_CODE_SYNC_ALREADY_RUNNING
-            elif status == "timeout":
-                self.logger.info("Status: timeout")
-                return self.EXIT_CODE_FINAL_STATUS_INCOMPLETE
-            elif status == "canceled":
-                self.logger.info("Status: canceled")
-                return self.EXIT_CODE_FINAL_STATUS_CANCELLED
-            elif status == "failed":
-                self.logger.info("Status: failed")
-                return self.EXIT_CODE_FINAL_STATUS_ERRORED
-            else:
-                self.logger.info(f"Status: {status}")
-                return self.EXIT_CODE_UNKNOWN_STATUS
-
-        elif response.status_code == 400:
-            self.logger.error(
-                f"There was an error when attempting to fetch the status of the job. The message returned from the API is {json['error']['errorString']}"
-            )
-            return self.EXIT_CODE_BAD_REQUEST
-
-        elif response.status_code == 401:
-            self.logger.error(
-                "Error occurred when attempting to authenticate, please ensure that the token provided is valid"
-            )
-            return self.EXIT_CODE_INVALID_CREDENTIALS
+        try:
+            response = self.get_run_status(run_counter)
+            json = response.json()
+            if response.status_code == 200:
+                status = json.get("runStatus")
+                # go through the statuses and return the appropriate exit code
+                # documentation does not provide options for status aside from completed
+                if status == "completed":
```

Comment:
consider using a dict instead of this many if statements. Personally I would've gone with 
```
statuses = {
     "completed":self.EXIT_CODE_FINAL_STATUS_COMPLETED,
     "pending": self.EXIT_CODE_FINAL_STATUS_PENDING,
     "running":self.EXIT_CODE_SYNC_ALREADY_RUNNING
     ...
if response.ok:
     status=json.get("runStatus")
     exit_code = statuses.get(status)
     if not exit_code:
          raise unknown status ... 
     return exit_code
else:
    ...     
```